File: WinForms/VB/SchedulerOutlookDemo/Modules/Common/Forms/ChangeViewForm.vb
Imports DevExpress.XtraEditors
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Navigation

Namespace DevExpress.XtraScheduler.Demos

    Public Partial Class ChangeViewForm
        Inherits XtraForm

        Private viewOptionsControl As ViewOptionsControl

        Private calendarOptionsControl As CalendarOptionsControl

        Private appointmentOptionsControl As AppointmentOptionsControl

        Public Sub New()
            InitializeComponent()
            accordionControl1.SelectedElement = accordionControlElement2
            viewOptionsControl = New ViewOptionsControl()
            calendarOptionsControl = New CalendarOptionsControl()
            appointmentOptionsControl = New AppointmentOptionsControl()
            viewOptionsControl.Dock = DockStyle.Fill
            calendarOptionsControl.Dock = DockStyle.Fill
            appointmentOptionsControl.Dock = DockStyle.Fill
            panelControl1.Controls.Add(viewOptionsControl)
            panelControl1.Controls.Add(calendarOptionsControl)
            panelControl1.Controls.Add(appointmentOptionsControl)
            UpdateCurrentOptionsVisibility()
        End Sub

        Public Sub LoadOptions(ByVal options As CommonOptions)
            viewOptionsControl.LoadOptions(options)
            calendarOptionsControl.LoadOptions(options)
            appointmentOptionsControl.LoadOptions(options)
        End Sub

        Public Sub SaveOptions(ByVal options As CommonOptions)
            viewOptionsControl.SaveOptions(options)
            calendarOptionsControl.SaveOptions(options)
            appointmentOptionsControl.SaveOptions(options)
        End Sub

        Private Sub accordionControl1_SelectedElementChanged(ByVal sender As Object, ByVal e As SelectedElementChangedEventArgs)
            If e.Element.Tag Is Nothing Then Return
            UpdateCurrentOptionsVisibility()
        End Sub

        Private Sub UpdateCurrentOptionsVisibility()
            Dim optionsName As String = accordionControl1.SelectedElement.Tag.ToString()
            SetOptionsControlVisible(viewOptionsControl, Equals(optionsName, "Views"))
            SetOptionsControlVisible(calendarOptionsControl, Equals(optionsName, "Calendars"))
            SetOptionsControlVisible(appointmentOptionsControl, Equals(optionsName, "Appointments"))
        End Sub

        Private Sub SetOptionsControlVisible(ByVal control As Control, ByVal value As Boolean)
            If control Is Nothing Then Return
            control.Visible = value
        End Sub
    End Class
End Namespace

File: WinForms/CS/SchedulerOutlookDemo/Modules/Common/Forms/ChangeViewForm.cs
using System;
using System.Linq;
using DevExpress.XtraEditors;
using System.Windows.Forms;
using DevExpress.XtraBars.Navigation;

namespace DevExpress.XtraScheduler.Demos {
    public partial class ChangeViewForm : XtraForm {
        ViewOptionsControl viewOptionsControl;
        CalendarOptionsControl calendarOptionsControl;
        AppointmentOptionsControl appointmentOptionsControl;

        public ChangeViewForm() {
            InitializeComponent();
            this.accordionControl1.SelectedElement = this.accordionControlElement2;
            this.viewOptionsControl = new ViewOptionsControl();
            this.calendarOptionsControl = new CalendarOptionsControl();
            this.appointmentOptionsControl = new AppointmentOptionsControl();
            this.viewOptionsControl.Dock = DockStyle.Fill;
            this.calendarOptionsControl.Dock = DockStyle.Fill;
            this.appointmentOptionsControl.Dock = DockStyle.Fill;
            this.panelControl1.Controls.Add(this.viewOptionsControl);
            this.panelControl1.Controls.Add(this.calendarOptionsControl);
            this.panelControl1.Controls.Add(this.appointmentOptionsControl);
            UpdateCurrentOptionsVisibility();
        }

        public void LoadOptions(CommonOptions options) {
            this.viewOptionsControl.LoadOptions(options);
            this.calendarOptionsControl.LoadOptions(options);
            this.appointmentOptionsControl.LoadOptions(options);
        }

        public void SaveOptions(CommonOptions options) {
            this.viewOptionsControl.SaveOptions(options);
            this.calendarOptionsControl.SaveOptions(options);
            this.appointmentOptionsControl.SaveOptions(options);
        }

        void accordionControl1_SelectedElementChanged(object sender, SelectedElementChangedEventArgs e) {
            if (e.Element.Tag == null)
                return;
            UpdateCurrentOptionsVisibility();
        }

        void UpdateCurrentOptionsVisibility() {
            string optionsName = this.accordionControl1.SelectedElement.Tag.ToString();
            SetOptionsControlVisible(this.viewOptionsControl, optionsName == "Views");
            SetOptionsControlVisible(this.calendarOptionsControl, optionsName == "Calendars");
            SetOptionsControlVisible(this.appointmentOptionsControl, optionsName == "Appointments");
        }

        void SetOptionsControlVisible(Control control, bool value) {
            if (control == null)
                return;
            control.Visible = value;
        }
    }
}
